hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/bulk/sort/SortOperator.java

```diff
@@ -89,7 +95,7 @@ public void open() throws Exception {
             binarySerializer,
             computer,
             comparator,
-            getContainingTask().getJobConfiguration());
+            conf);
```

Comment:
Just curious why we have to change this line now?

Reply:
getContainingTask().getJobConfiguration() returns task configuration which does not have the table exec options required by the BinaryExternalSorter. [FLINK-30989](https://issues.apache.org/jira/browse/FLINK-30989) has changed it to the Flink Client Configuration in the [SortOperator](https://github.com/apache/flink/pull/22024/files#diff-1ccf2a96b98cbf22d3df6968082ccb3087d575f135828610ed57be55e4264cddL88).

Reply:
So after this fix, user expects to pass around the sort options through SQL (guess these options are configured in `flink-conf.yaml` or in SqlClient session)?  In `HoodieTableFactory#` we can also fetch a configuration through `context.getConfiguration()`, should we use that configuration then?

Reply:
Yes right, the config needs to be set in flink-conf.yaml or SqlClient session. https://nightlies.apache.org/flink/flink-docs-master/docs/dev/table/config/

Both Context.getConfiguration and the configuration used above are the same Flink Configuration (flink-conf.yaml) with the table options. The later needs less code changes, at below classes to pass the configuration object

```
 Pipelines#bulkInsert -> SortOperatorGen#createSortOperator -> SortOperator#init
```

whereas former needs changes at below classes to pass the Context.getConfiguration

```
HoodieTableSink#getSinkRuntimeProvider -> Pipelines#bulkInsert -> SortOperatorGen#createSortOperator -> SortOperator#init
```
> should we use that configuration then?

I think not required.

Reply:
I'm afraid the fix is not right, take a look at the Flink fix: https://github.com/apache/flink/pull/22024, they fetch the options through the planner table config which is passed around during plan generation, and here, tha options from the catalog table does not incude any table config options from `flink-conf.yaml` or `SqlClient` session, it's just the SQL options.

Reply:
You are right; the configuration used to instantiate HoodieTableSink had only the default Flink configuration values with the SQL options. The DynamicTableFactory#context is the right one to use and is from the Planner, which has the configs specified from both flink-conf.yaml and SqlClient Session, with later taking first precedence. I have tested below to verify the new changes with configuration from the context working as expected.

1. Add table.exec.spill-compression.block-size: 40 m in /etc/flink/conf/flink-conf.yaml
2. Create a hudi table in flink sql-client session with 'write.operation' = 'bulk_insert'.
3. Insert some rows into the table.
4. taskmanager.log shows below with right compressionBlockSize.

```
taskmanager.log:2023-04-26 09:26:43,130 INFO  org.apache.flink.table.runtime.operators.sort.BinaryExternalSorter [] - BinaryExternalSorter with initial memory segments 16384, maxNumFileHandles(128), compressionEnable(true), compressionCodecFactory(class org.apache.flink.runtime.io.compression.Lz4BlockCompressionFactory), compressionBlockSize(41943040).
```
5. Set below and run the insert command and taskmanager.log shows the overridden compressionBlockSize

```
Flink SQL> set 'table.exec.spill-compression.block-size' = '80 m';

taskmanager.log:2023-04-26 09:29:14,906 INFO  org.apache.flink.table.runtime.operators.sort.BinaryExternalSorter [] - BinaryExternalSorter with initial memory segments 16384, maxNumFileHandles(128), compressionEnable(true), compressionCodecFactory(class org.apache.flink.runtime.io.compression.Lz4BlockCompressionFactory), compressionBlockSize(83886080).
```